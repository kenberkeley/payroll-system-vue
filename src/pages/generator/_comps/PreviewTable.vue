<template>
  <table class="table is-fullwidth -preview-table">
    <thead class="has-text-weight-medium">
      <tr>
        <td>{{ titles[0] }}</td>
        <td>{{ titles[1] }}</td>
      </tr>
    </thead>
    <tbody>
      <tr v-for="(item, idx) in rows" :key="idx">
        <td>{{ item[0] }}</td>
        <td>{{ item[2] }} {{ item[1] }}</td>
      </tr>
      <tr></tr>
    </tbody>
  </table>
</template>
<script>
export default {
  props: {
    /** [<key>, <val>, <val's prefix>?] */
    titles: { type: Array, required: true },
    rows: { type: Array, required: true }
  }
}
</script>
<style lang="scss">
// refer to https://stackoverflow.com/a/40213845/5172890
.-preview-table {
  tr th:first-child,
  tr td:first-child {
    width: 18%;
    min-width: 8.5em;
  }
}
</style>
