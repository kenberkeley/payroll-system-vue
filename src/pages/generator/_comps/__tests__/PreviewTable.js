/* global it, expect */
import { mount } from '@vue/test-utils'
import PreviewTable from '../PreviewTable.vue'

it('Generator PreviewTable', () => {
  const wrapper = mount(PreviewTable, {
    propsData: {
      titles: ['T1', 'T2'],
      rows: [
        ['A', 'B'],
        ['C', 'D'],
        ['E', 'F', '$']
      ]
    }
  })
  expect(wrapper.element).toMatchSnapshot()
})
