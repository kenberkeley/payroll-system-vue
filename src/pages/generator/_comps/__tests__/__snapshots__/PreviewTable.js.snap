// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator PreviewTable 1`] = `
<table
  class="table is-fullwidth -preview-table"
>
  <thead
    class="has-text-weight-medium"
  >
    <tr>
      <td>
        T1
      </td>
       
      <td>
        T2
      </td>
    </tr>
  </thead>
   
  <tbody>
    <tr>
      <td>
        A
      </td>
       
      <td>
         B
      </td>
    </tr>
    <tr>
      <td>
        C
      </td>
       
      <td>
         D
      </td>
    </tr>
    <tr>
      <td>
        E
      </td>
       
      <td>
        $ F
      </td>
    </tr>
     
    <tr />
  </tbody>
</table>
`;
